x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/trial_license_complete_tier/execution_logic/eql_alert_suppression.ts

```diff
@@ -1786,40 +1793,1230 @@ export default ({ getService }: FtrProviderContext) => {
+        // host.name value as sequence alert 1
+
+        await indexListOfSourceDocuments([doc1, doc1WithLaterTimestamp, doc2WithLaterTimestamp]);
+
+        const rule: EqlRuleCreateProps = {
+          ...getEqlRuleForAlertTesting(['ecs_compliant']),
+          query: getSequenceQuery(id),
+          alert_suppression: {
+            group_by: ['host.name'],
+            missing_fields_strategy: 'suppress',
+          },
+          from: 'now-35m',
+          interval: '30m',
+        };
+
+        const { previewId } = await previewRule({
+          supertest,
+          rule,
+          timeframeEnd: new Date('2020-10-28T07:00:00.000Z'),
+          invocationCount: 1,
+        });
+        const previewAlerts = await getPreviewAlerts({
+          es,
+          previewId,
+          sort: [ALERT_ORIGINAL_TIME],
+        });
+        // we expect one alert and two suppressed alerts
+        // and two building block alerts, let's confirm that
+        expect(previewAlerts.length).toEqual(3);
+        const [sequenceAlert, buildingBlockAlerts] = partition(
+          previewAlerts,
+          (alert) => alert?._source?.['kibana.alert.building_block_type'] == null
+        );
+        expect(buildingBlockAlerts.length).toEqual(2);
+        expect(sequenceAlert.length).toEqual(1);
+
+        expect(sequenceAlert[0]?._source).toEqual({
+          ...sequenceAlert[0]?._source,
+          [ALERT_SUPPRESSION_TERMS]: [
+            {
+              field: 'host.name',
+              value: ['host-a'],
+            },
+          ],
+          [TIMESTAMP]: '2020-10-28T07:00:00.000Z',
+          [ALERT_LAST_DETECTED]: '2020-10-28T07:00:00.000Z',
+          [ALERT_SUPPRESSION_START]: laterTimestamp,
+          [ALERT_SUPPRESSION_END]: laterTimestamp2,
+          [ALERT_SUPPRESSION_DOCS_COUNT]: 1,
         });
+      });
+
+      it('suppresses alerts in a given rule execution when a subsequent event for an sequence has the suppression field undefined', async () => {
+        const id = uuidv4();
+        const timestamp = '2020-10-28T06:50:00.000Z';
+        const laterTimestamp = '2020-10-28T06:51:00.000Z';
+        const laterTimestamp2 = '2020-10-28T06:53:00.000Z';
+        const doc1 = {
+          id,
+          '@timestamp': timestamp,
+          host: { name: 'host-a' },
+        };
+        const doc1WithLaterTimestamp = {
+          ...doc1,
+          '@timestamp': laterTimestamp,
+        };
+
+        const doc2WithNoHost = {
+          ...doc1,
+          '@timestamp': laterTimestamp2,
+          host: undefined,
+        };
+
+        // sequence alert 1 will be doc1 and doc1WithLaterTimestamp
+        // sequence alert 2 will be doc1WithLaterTimestamp and doc2WithNoHost
+        // the reason for the second alert is because despite the value being null
+        // in one of the two events in the sequence, the sequence alert will
+        // adopt the value for host.name and be suppressible.
```

Comment:
@dhurley14 
I did additional testing here and it seems something is off.

I disabled suppression and got 2 generated alerts:

```JSON
[
  {
    "_index": ".internal.preview.alerts-security.alerts-default-000001",
    "_id": "0fdbae996c6ebb103245b671ab86a279880332913cc415883d694abd97cb8986",
    "_score": null,
    "_source": {
      "kibana.alert.rule.execution.timestamp": "2024-10-10T12:35:26.394Z",
      "kibana.alert.start": "2024-10-10T12:35:26.394Z",
      "kibana.alert.last_detected": "2024-10-10T12:35:26.394Z",
      "kibana.alert.intended_timestamp": "2020-10-28T07:00:00.000Z",
      "kibana.version": "9.0.0",
      "kibana.alert.rule.parameters": {
        "description": "Tests a simple query",
        "risk_score": 1,
        "severity": "high",
        "author": [],
        "false_positives": [],
        "from": "now-35m",
        "rule_id": "eql-rule",
        "max_signals": 100,
        "risk_score_mapping": [],
        "severity_mapping": [],
        "threat": [],
        "to": "now",
        "references": [],
        "version": 1,
        "exceptions_list": [],
        "immutable": false,
        "rule_source": {
          "type": "internal"
        },
        "related_integrations": [],
        "required_fields": [],
        "setup": "",
        "type": "eql",
        "language": "eql",
        "index": [
          "ecs_compliant"
        ],
        "query": "sequence [any where id == \"8dbd1368-a2a9-4d49-b442-d6f3e89f10cb\"] [any where id == \"8dbd1368-a2a9-4d49-b442-d6f3e89f10cb\"]"
      },
      "kibana.alert.rule.category": "Event Correlation Rule",
      "kibana.alert.rule.consumer": "siem",
      "kibana.alert.rule.execution.uuid": "db59bba8-6635-412a-9d0a-bdd127451b42",
      "kibana.alert.rule.name": "Alert Testing Query",
      "kibana.alert.rule.producer": "preview-producer",
      "kibana.alert.rule.revision": 0,
      "kibana.alert.rule.rule_type_id": "siem.eqlRule",
      "kibana.alert.rule.uuid": "bdbd47ce-86f6-4e92-9e6c-3fa5b04bafc3",
      "kibana.space_ids": [
        "default"
      ],
      "kibana.alert.rule.tags": [],
      "@timestamp": "2020-10-28T07:00:00.000Z",
      "id": "8dbd1368-a2a9-4d49-b442-d6f3e89f10cb",
      "host": {
        "name": "host-a"
      },
      "event.kind": "signal",
      "kibana.alert.ancestors": [
        {
          "id": "R2BtdpIBVlvnLm_0RSkp",
          "type": "event",
          "index": "ecs_compliant",
          "depth": 0
        },
        {
          "id": "9b29c6dd219631e90c819e21545c52d0c54a2b98c8542d441856799fb54dbeb1",
          "type": "signal",
          "index": "",
          "depth": 1,
          "rule": "bdbd47ce-86f6-4e92-9e6c-3fa5b04bafc3"
        },
        {
          "id": "SGBtdpIBVlvnLm_0RSkp",
          "type": "event",
          "index": "ecs_compliant",
          "depth": 0
        },
        {
          "id": "9b29a815e10fc5a48cfd174998bea2c5f7a86a3f1fbd81884961f54991c6e912",
          "type": "signal",
          "index": "",
          "depth": 1,
          "rule": "bdbd47ce-86f6-4e92-9e6c-3fa5b04bafc3"
        }
      ],
      "kibana.alert.status": "active",
      "kibana.alert.workflow_status": "open",
      "kibana.alert.depth": 2,
      "kibana.alert.reason": "event on host-a created high alert Alert Testing Query.",
      "kibana.alert.severity": "high",
      "kibana.alert.risk_score": 1,
      "kibana.alert.rule.actions": [],
      "kibana.alert.rule.author": [],
      "kibana.alert.rule.created_at": "2024-10-10T12:35:25.726Z",
      "kibana.alert.rule.created_by": "elastic",
      "kibana.alert.rule.description": "Tests a simple query",
      "kibana.alert.rule.enabled": true,
      "kibana.alert.rule.exceptions_list": [],
      "kibana.alert.rule.false_positives": [],
      "kibana.alert.rule.from": "now-35m",
      "kibana.alert.rule.immutable": false,
      "kibana.alert.rule.interval": "30m",
      "kibana.alert.rule.indices": [
        "ecs_compliant"
      ],
      "kibana.alert.rule.max_signals": 100,
      "kibana.alert.rule.references": [],
      "kibana.alert.rule.risk_score_mapping": [],
      "kibana.alert.rule.rule_id": "eql-rule",
      "kibana.alert.rule.severity_mapping": [],
      "kibana.alert.rule.threat": [],
      "kibana.alert.rule.to": "now",
      "kibana.alert.rule.type": "eql",
      "kibana.alert.rule.updated_at": "2024-10-10T12:35:25.726Z",
      "kibana.alert.rule.updated_by": "elastic",
      "kibana.alert.rule.version": 1,
      "kibana.alert.workflow_tags": [],
      "kibana.alert.workflow_assignee_ids": [],
      "kibana.alert.rule.risk_score": 1,
      "kibana.alert.rule.severity": "high",
      "kibana.alert.original_time": "2020-10-28T07:00:00.000Z",
      "kibana.alert.uuid": "0fdbae996c6ebb103245b671ab86a279880332913cc415883d694abd97cb8986",
      "kibana.alert.group.id": "0fdbae996c6ebb103245b671ab86a279880332913cc415883d694abd97cb8986"
    },
    "sort": [
      1603868400000
    ]
  },
  {
    "_index": ".internal.preview.alerts-security.alerts-default-000001",
    "_id": "cc5041c070f1e56f0be7e88001e662fb27666d0184452c49b296d437c1ee6848",
    "_score": null,
    "_source": {
      "kibana.alert.rule.execution.timestamp": "2024-10-10T12:35:26.394Z",
      "kibana.alert.start": "2024-10-10T12:35:26.394Z",
      "kibana.alert.last_detected": "2024-10-10T12:35:26.394Z",
      "kibana.alert.intended_timestamp": "2020-10-28T07:00:00.000Z",
      "kibana.version": "9.0.0",
      "kibana.alert.rule.parameters": {
        "description": "Tests a simple query",
        "risk_score": 1,
        "severity": "high",
        "author": [],
        "false_positives": [],
        "from": "now-35m",
        "rule_id": "eql-rule",
        "max_signals": 100,
        "risk_score_mapping": [],
        "severity_mapping": [],
        "threat": [],
        "to": "now",
        "references": [],
        "version": 1,
        "exceptions_list": [],
        "immutable": false,
        "rule_source": {
          "type": "internal"
        },
        "related_integrations": [],
        "required_fields": [],
        "setup": "",
        "type": "eql",
        "language": "eql",
        "index": [
          "ecs_compliant"
        ],
        "query": "sequence [any where id == \"8dbd1368-a2a9-4d49-b442-d6f3e89f10cb\"] [any where id == \"8dbd1368-a2a9-4d49-b442-d6f3e89f10cb\"]"
      },
      "kibana.alert.rule.category": "Event Correlation Rule",
      "kibana.alert.rule.consumer": "siem",
      "kibana.alert.rule.execution.uuid": "db59bba8-6635-412a-9d0a-bdd127451b42",
      "kibana.alert.rule.name": "Alert Testing Query",
      "kibana.alert.rule.producer": "preview-producer",
      "kibana.alert.rule.revision": 0,
      "kibana.alert.rule.rule_type_id": "siem.eqlRule",
      "kibana.alert.rule.uuid": "bdbd47ce-86f6-4e92-9e6c-3fa5b04bafc3",
      "kibana.space_ids": [
        "default"
      ],
      "kibana.alert.rule.tags": [],
      "@timestamp": "2020-10-28T07:00:00.000Z",
      "id": "8dbd1368-a2a9-4d49-b442-d6f3e89f10cb",
      "event.kind": "signal",
      "kibana.alert.ancestors": [
        {
          "id": "SGBtdpIBVlvnLm_0RSkp",
          "type": "event",
          "index": "ecs_compliant",
          "depth": 0
        },
        {
          "id": "e655ebc2c0f7050f579df539f44229a40e14af11c4383e5d171bedff8d98146c",
          "type": "signal",
          "index": "",
          "depth": 1,
          "rule": "bdbd47ce-86f6-4e92-9e6c-3fa5b04bafc3"
        },
        {
          "id": "SWBtdpIBVlvnLm_0RSkp",
          "type": "event",
          "index": "ecs_compliant",
          "depth": 0
        },
        {
          "id": "0bdf3f44443845841fa95c0998cb742d8abda0ebdccf3c78e3b0d97742624319",
          "type": "signal",
          "index": "",
          "depth": 1,
          "rule": "bdbd47ce-86f6-4e92-9e6c-3fa5b04bafc3"
        }
      ],
      "kibana.alert.status": "active",
      "kibana.alert.workflow_status": "open",
      "kibana.alert.depth": 2,
      "kibana.alert.severity": "high",
      "kibana.alert.risk_score": 1,
      "kibana.alert.rule.actions": [],
      "kibana.alert.rule.author": [],
      "kibana.alert.rule.created_at": "2024-10-10T12:35:25.726Z",
      "kibana.alert.rule.created_by": "elastic",
      "kibana.alert.rule.description": "Tests a simple query",
      "kibana.alert.rule.enabled": true,
      "kibana.alert.rule.exceptions_list": [],
      "kibana.alert.rule.false_positives": [],
      "kibana.alert.rule.from": "now-35m",
      "kibana.alert.rule.immutable": false,
      "kibana.alert.rule.interval": "30m",
      "kibana.alert.rule.indices": [
        "ecs_compliant"
      ],
      "kibana.alert.rule.max_signals": 100,
      "kibana.alert.rule.references": [],
      "kibana.alert.rule.risk_score_mapping": [],
      "kibana.alert.rule.rule_id": "eql-rule",
      "kibana.alert.rule.severity_mapping": [],
      "kibana.alert.rule.threat": [],
      "kibana.alert.rule.to": "now",
      "kibana.alert.rule.type": "eql",
      "kibana.alert.rule.updated_at": "2024-10-10T12:35:25.726Z",
      "kibana.alert.rule.updated_by": "elastic",
      "kibana.alert.rule.version": 1,
      "kibana.alert.workflow_tags": [],
      "kibana.alert.workflow_assignee_ids": [],
      "kibana.alert.rule.risk_score": 1,
      "kibana.alert.rule.severity": "high",
      "kibana.alert.original_time": "2020-10-28T07:00:00.000Z",
      "kibana.alert.reason": "event created high alert Alert Testing Query.",
      "kibana.alert.uuid": "cc5041c070f1e56f0be7e88001e662fb27666d0184452c49b296d437c1ee6848",
      "kibana.alert.group.id": "cc5041c070f1e56f0be7e88001e662fb27666d0184452c49b296d437c1ee6848"
    },
    "sort": [
      1603868400000
    ]
  }
]
```


This is expected. Also expected to see host.name:host-a in first shell alert and no host.name in second.

But when suppression enabled, second alert is getting suppressed. But it should not, since they don't have common `host.name` field